### PR TITLE
Bump to Dart SDK 1.24.0-dev.6.7.

### PR DESCRIPTION
**Pull request by pq**
Keeping up with the Joneses! 👍 

https://github.com/dart-lang/sdk/blob/1.24.0-dev.6.7/CHANGELOG.md

cc @tvolkert @Hixie

### CODE REVIEW

File: dev/devicelab/pubspec.yaml [hunk 1]
```diff
@@ -17,4 +17,4 @@ dependencies:
 
 dev_dependencies:
   # See packages/flutter_test/pubspec.yaml for why we're pinning this version.
-  test: 0.12.20
+  test: 0.12.20+12
```
Comment (tvolkert) [inline dev/devicelab/pubspec.yaml]: We're on `0.12.21` internally - can we bump this to that version without problems?

File: packages/flutter_tools/lib/src/commands/analyze_once.dart [hunk 2]
```diff
@@ -31,6 +31,9 @@ class AnalyzeOnce extends AnalyzeBase {
   /// The working directory for testing analysis using dartanalyzer
   final Directory workingDirectory;
 
+  /// Packages whose source is defined in the vended SDK. 
+  static final List<String> _vendedSdkPackages = <String>['analyzer', 'front_end', 'kernel'];
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/commands/analyze_once.dart]: const?

File: dev/devicelab/pubspec.yaml (hunk 1 above)
Comment (pq) [inline dev/devicelab/pubspec.yaml]: Oh.  Probably!  I'll give it a whirl.  Thanks for the catch.

File: packages/flutter_tools/lib/src/commands/analyze_once.dart (hunk 2 above)
Comment (pq) [inline packages/flutter_tools/lib/src/commands/analyze_once.dart]: :) I'd just changed that locally.  Beat me to it!

File: packages/flutter_tools/lib/src/commands/analyze_once.dart [hunk 3]
```diff
@@ -158,10 +161,10 @@ class AnalyzeOnce extends AnalyzeBase {
             if (colon > 0) {
               final String packageName = line.substring(0, colon);
               final String packagePath = fs.path.fromUri(line.substring(colon+1));
-              // Ensure that we only add the `analyzer` package defined in the vended SDK (and referred to with a local fs.path. directive).
-              // Analyzer package versions reached via transitive dependencies (e.g., via `test`) are ignored since they would produce
-              // spurious conflicts.
-              if (packageName != 'analyzer' || packagePath.startsWith('..'))
+              // Ensure that we only add  `analyzer` and dependent packages defined in the vended SDK (and referred to with a local
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/commands/analyze_once.dart]: Nit: there's a double space here

Comment (pq) [inline packages/flutter_tools/lib/src/commands/analyze_once.dart]: Good eye! 🦅

**Review by tvolkert — APPROVED** [review]
LGTM.  Thanks!

**Pull request merged**